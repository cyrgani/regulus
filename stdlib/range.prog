# Returns a list of all integers from `start` to `end`, including `start` but not `end`.
# Rust equivalent: `start..end`
def(range, start, end, _(
    if(>(start, end), error("cannot construct range with start > end")),
    =(i, start),
    =(l, list()),
    while(<(i, end), _(
        =(l, push(l, i)),
        =(i, +(i, 1)),
    )),
    l
)),

=(.., range)