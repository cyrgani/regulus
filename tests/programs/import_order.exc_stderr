NameError: No function `range` found!

