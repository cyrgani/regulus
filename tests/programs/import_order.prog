assert_eq(catch(range(0, 1)), "Name: No function `range` found!"),
import("range")
