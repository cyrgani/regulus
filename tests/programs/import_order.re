__builtin_print_catch(new_fraction(0, 1)),
import(fraction)
