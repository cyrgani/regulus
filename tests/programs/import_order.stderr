NameError: No function `new_fraction` found!
at programs/import_order.re:1:35
at programs/import_order.re:1:22
